### DISCUSSION TITLE
Filter expression (Multi-Conditions) by HTTP REST API

### DISCUSSION THREAD

**Discussion by imsgao** [pocketbase/pocketbase#5132] [Q&A] [answered]
Version: [PocketBase v0.22.13](https://github.com/pocketbase/pocketbase/releases)
Operation System: Debian 11

**Success Case**: return 200 when using single field  filter, like: 
`?filter=(created>'2022-01-01')`

**Failure Case**: return 400 when using multi-fields filter, like:
`?filter=(title~'abc' && created>'2022-01-01')`

Error Message:
`
{
"code": 400,
"message": "Something went wrong while processing your request.",
"data": {}
}
`

I am using pocketbase as app backend, so implemented api request with native Swift and  Alamofire.
❤️ Thank you so much for helping to boost my start-up project."

**Comment by ganigeorgiev** [comment, marked as answer]
The easiest way to rule out PocketBase issue is to try the filter in the Admin UI.

If you are constructing the request manually, the `filter` query parameter must be properly URL encoded, aka. in your case the valid query parameter should be:

```js
?filter=title~%27abc%27%20%26%26%20created%3E%272022-01-01%27
```

Also note that the full-date time format is `Y-m-d H:i:s.uZ`, so a more precise date constraint could be:
```js
// adjust the date depending on your requirement
?filter=title~'abc' && created>='2022-01-01 00:00:00.000Z'

// url encoded version:
?filter=title~%27abc%27%20%26%26%20created%3E%3D%272022-01-01%2000%3A00%3A00.000Z%27
```

For more detailed error messages you could also start the executable with `--dev` and/or find the failed request in the _Admin UI > Logs_.

If the above doesn't help, please export your collections configuration as json (_Admin UI > Settings > Export collections_) and provide a minimal code sample how you are sending request.

**Reply by imsgao** [reply]
Thank you for the anwser, it works!
My mistake is added parentheses to query params.
